### PR TITLE
option to configure agent sds

### PR DESCRIPTION
**Pull request by grameshtwilio**
Not sure if this is the preferred way. Could also populate the values file with defaults from https://github.com/spiffe/spire/blob/main/doc/spire_agent.md#sds-configuration

### CODE REVIEW

**Comment by faisal-memon** [comment]
Hi @grameshtwilio thanks for the submission. Populating the values with defaults is preferred, but we understand people are busy. If you are able to do that, then great. If not we can take the changes as is and fill them in later.

**Comment by faisal-memon** [comment]
Also note we have a [major docs refactor](https://github.com/spiffe/helm-charts/pull/431) we are hoping to merge next. There might be some merge conflicts here when you update after than one is merged.

**Comment by grameshtwilio** [comment]
> Also note we have a [major docs refactor](https://github.com/spiffe/helm-charts/pull/431) we are hoping to merge next. There might be some merge conflicts here when you update after than one is merged.

I'll keep an eye on it and rebase/alter this PR to incorporate the new format.

**Review by faisal-memon — APPROVED** [review]
Thanks @grameshtwilio

File: charts/spire/README.md [hunk 1]
```diff
@@ -332,6 +332,10 @@ Now you can interact with the Spire agent socket from your own application. The
 | spire-agent.readinessProbe.initialDelaySeconds | int | `15` | Initial delay seconds for readinessProbe |
 | spire-agent.readinessProbe.periodSeconds | int | `60` | Period seconds for readinessProbe |
 | spire-agent.resources | object | `{}` |  |
+| spire-agent.sds.default_all_bundles_name | string | `"ALL"` | The Validation Context resource name to use for all bundles (including federated) with Envoy SDS |
+| spire-agent.sds.default_bundle_name | string | `"ROOTCA"` | The Validation Context resource name to use for the default X.509 bundle with Envoy SDS |
+| spire-agent.sds.default_svid_name | string | `"default"` | The TLS Certificate resource name to use for the default X509-SVID with Envoy SDS |
+| spire-agent.sds.disable_spiffe_cert_validation | bool | `false` | Disable Envoy SDS custom validation |
```
Comment (kfox1111) [inline charts/spire/README.md:338]: These should be updated to follow the convention of the chart. See this for details:
https://github.com/spiffe/helm-charts/blob/main/project/conventions.md#preferred-characters-for-helm-path-elements

**Review by kfox1111 — COMMENTED** [review]
Looking good. Some naming issues inline.

File: charts/spire/README.md (hunk 1 above)
Comment (grameshtwilio) [reply charts/spire/README.md:338]: @kfox1111 , are you referring to the preference for camel case?

Comment (grameshtwilio) [reply charts/spire/README.md]: I think this is addressed

File: charts/spire/charts/spire-agent/templates/configmap.yaml [hunk 2]
```diff
@@ -12,6 +12,11 @@ agent:
   trust_bundle_path: "/run/spire/bundle/bundle.crt"
   {{- end }}
   trust_domain: {{ include "spire-lib.trust-domain" . | quote }}
+  sds:
+    default_svid_name: {{ .Values.sds.defaultSvidName | quote }}
+    default_bundle_name: {{ .Values.sds.defaultBundleName | quote }}
+    default_all_bundles_name: {{ .Values.sds.defaultAllBundlesName | quote }}
+    disable_spiffe_cert_validation: {{ .Values.sds.disableSpiffeCertValidation }}
```
Comment (marcofranssen) [inline charts/spire/charts/spire-agent/templates/configmap.yaml:19]: Below you are stating this is an optional configuration, however here in the implementation it isn't.

Maybe we should add a enabled flag to conditionally enable or disable these settings.

File: charts/spire/charts/spire-agent/values.yaml [hunk 3]
```diff
@@ -170,6 +170,17 @@ workloadAttestors:
     ## @param workloadAttestors.k8s.disableContainerSelectors Set to true if using holdApplicationUntilProxyStarts in Istio
     disableContainerSelectors: false
 
+# Optional SDS configuration section.
+sds:
```
Comment (marcofranssen) [inline charts/spire/charts/spire-agent/values.yaml:174]:
```suggestion
sds:
  ## @param sds.enabled Enables Envoy SDS configuration
  enabled: false
```

Don't forget to regenerate the docs.

**Review by marcofranssen — CHANGES_REQUESTED** [review]
Suggestion on making the sds config really optional.

File: charts/spire/charts/spire-agent/values.yaml [hunk 4]
```diff
@@ -170,6 +170,20 @@ workloadAttestors:
     ## @param workloadAttestors.k8s.disableContainerSelectors Set to true if using holdApplicationUntilProxyStarts in Istio
     disableContainerSelectors: false
 
+sds:
+  ## @param sds.enabled Enables Envoy SDS configuration
+  enabled: false
+  ## @param sds.enabled enables setting non-default values for the SDS configuration
+  enabled: false
```
Comment (marcofranssen) [inline charts/spire/charts/spire-agent/values.yaml:177]:
```suggestion
```

**Review by marcofranssen — APPROVED** [review]
LGTM 🚀

**Pull request merged**